modules/installation/pages/upgrade.adoc

```diff
@@ -85,6 +85,136 @@ $ gadmin restart all -y
 
 . Enable CRR on the DR cluster.
 
+[#_rolling_upgrade]
+== Rolling Upgrade (Preview)
+
+A rolling upgrade allows you to upgrade your TigerGraph installation with minimal downtime by upgrading nodes in batches, ensuring that at least one full replica of the cluster is online throughout the process. 
+
+[NOTE]
+====
+Rolling Upgrade is only available for HA clusters (High Availability) and when making a maintenance update (e.g., 4.2.0 to 4.2.1, 4.3.1 to 4.3.3). 
+Upgrading from 4.2.0 to 4.3.0 or any other major version is not supported.
+====
+
+During a rolling upgrade, several operations are not permitted to ensure a smooth upgrade process. 
+These operations include installing queries, running schema changes, dropping all data, using `gsql --reset`, creating graphs, performing backup/restore operations, clearing the graph store, exporting/importing graphs, installing functions, changing TigerGraph configurations, starting or stopping services, setting up CRR (though CRR will continue working if already set up), running `gadmin init` for cluster initialization, upgrading services with `gadmin upgrade`, and performing service resets with `gadmin reset`. 
+While the rolling upgrade is in progress, loading jobs and running queries will continue to function; however, queries requiring all replicas to be online may fail, in which case you can retry them in a round-robin manner. 
+Please note that occasional drops in Queries Per Second (QPS) may occur during the upgrade. Additionally, services like Zookeeper (ZK), Kafka, and ETCD will not be upgraded during this process.
```

Comment:
Let's not use the word "drop" if we only mean a "slowdown". 
"Drop" usually has a more serious meaning with databases.